src/marqo/tensor_search/validation.py

```diff
@@ -482,3 +527,20 @@ def validate_delete_docs_request(delete_request: MqDeleteDocsRequest, max_delete
         validate_id(_id)
 
     return delete_request
+
+
+def validate_result_count(result_count: int):
+    """Validates the result count (limit) for a search operation.
+
+    Args:
+        result_count: the result count to validate
+    Returns:
+        result_count, if nothing is raised
+    """
+    if not isinstance(result_count, int):
+        raise IllegalRequestedDocCount(f"result_count must be an int! Received {result_count} of type {type(result_count)}")
+
+    if result_count <= 0:
```

Comment:
I think the code will be more readable if you simply have this in `tensor_search.search` especially after removing the type check above